### DISCUSSION TITLE
kubernetes-dashbord: no endpoints available for service \"https:kubernetes-dashboard:\"

### DISCUSSION THREAD

**Discussion by ivdok** [k3s-io/k3s#7243] [Q&A]
**Environmental Info:**
K3s Version: 
k3s version v1.26.3+k3s1 (01ea3ff2)
go version go1.19.7

Node(s) CPU architecture, OS, and Version: 
Linux k8s 6.2.6-arch1-1 #1 SMP PREEMPT_DYNAMIC Mon, 13 Mar 2023 17:02:08 +0000 x86_64 GNU/Linux

Cluster Configuration:
Single-node deployment

**Describe the bug:**
While following https://docs.k3s.io/installation/kube-dashboard after installation, link http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/ returns this JSON:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "no endpoints available for service \"https:kubernetes-dashboard:\"",
  "reason": "ServiceUnavailable",
  "code": 503
}
```
Port 8001 was forwarded via SSH to local PC.

**Steps To Reproduce:**
curl -sfL https://get.k3s.io | sh -
GITHUB_URL=https://github.com/kubernetes/dashboard/releases
VERSION_KUBE_DASHBOARD=$(curl -w '%{url_effective}' -I -L -s -S ${GITHUB_URL}/latest -o /dev/null | sed -e 's|.*/||')
k3s kubectl create -f https://raw.githubusercontent.com/kubernetes/dashboard/${VERSION_KUBE_DASHBOARD}/aio/deploy/recommended.yaml
k3s kubectl create -f dashboard.yml ## two manifests merged into one file with ---

**Expected behavior:**
Dashboard is opened in web browser

**Actual behavior:**
JSON with nondescript error

**Comment by ivdok** [comment]
[root@k8s ~]# k3s check-config

Verifying binaries in /var/lib/rancher/k3s/data/c26e7571d760c5f199d18efd197114f1ca4ab1e6ffe494f96feb65c87fcb8cf0/bin:
- sha256sum: good
- links: good

System:
- /usr/bin iptables v1.8.9 (legacy): ok
- swap: disabled
- routes: ok

Limits:
- /proc/sys/kernel/keys/root_maxkeys: 1000000

info: reading kernel config from /proc/config.gz ...

Generally Necessary:
- cgroup hierarchy: cgroups V2 mounted, cpu|cpuset|memory controllers status: good
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MEMCG: enabled
- CONFIG_KEYS: enabled
- CONFIG_VETH: enabled (as module)
- CONFIG_BRIDGE: enabled (as module)
- CONFIG_BRIDGE_NETFILTER: enabled (as module)
- CONFIG_IP_NF_FILTER: enabled (as module)
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_IPVS: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_COMMENT: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_MULTIPORT: enabled (as module)
- CONFIG_IP_NF_NAT: enabled (as module)
- CONFIG_NF_NAT: enabled (as module)
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_USER_NS: enabled
- CONFIG_SECCOMP: enabled
- CONFIG_CGROUP_PIDS: enabled
- CONFIG_BLK_CGROUP: enabled
- CONFIG_BLK_DEV_THROTTLING: enabled
- CONFIG_CGROUP_PERF: enabled
- CONFIG_CGROUP_HUGETLB: enabled
- CONFIG_NET_CLS_CGROUP: enabled (as module)
- CONFIG_CGROUP_NET_PRIO: enabled
- CONFIG_CFS_BANDWIDTH: enabled
- CONFIG_FAIR_GROUP_SCHED: enabled
- CONFIG_RT_GROUP_SCHED: missing
- CONFIG_IP_NF_TARGET_REDIRECT: enabled (as module)
- CONFIG_IP_SET: enabled (as module)
- CONFIG_IP_VS: enabled (as module)
- CONFIG_IP_VS_NFCT: enabled
- CONFIG_IP_VS_PROTO_TCP: enabled
- CONFIG_IP_VS_PROTO_UDP: enabled
- CONFIG_IP_VS_RR: enabled (as module)
- CONFIG_EXT4_FS: enabled (as module)
- CONFIG_EXT4_FS_POSIX_ACL: enabled
- CONFIG_EXT4_FS_SECURITY: enabled
- Network Drivers:
  - "overlay":
    - CONFIG_VXLAN: enabled (as module)
      Optional (for encrypted networks):
      - CONFIG_CRYPTO: enabled
      - CONFIG_CRYPTO_AEAD: enabled (as module)
      - CONFIG_CRYPTO_GCM: enabled (as module)
      - CONFIG_CRYPTO_SEQIV: enabled (as module)
      - CONFIG_CRYPTO_GHASH: enabled (as module)
      - CONFIG_XFRM: enabled
      - CONFIG_XFRM_USER: enabled
      - CONFIG_XFRM_ALGO: enabled
      - CONFIG_INET_ESP: enabled (as module)
      - CONFIG_INET_XFRM_MODE_TRANSPORT: missing
- Storage Drivers:
  - "overlay":
    - CONFIG_OVERLAY_FS: enabled (as module)

STATUS: pass

**Reply by brandond** [reply]
> ` "message": "no endpoints available for service \"https:kubernetes-dashboard:\"",`
>
> JSON with nondescript error

That's not a nondescript error, it is telling you specifically what the problem is - there are no endpoints for the service.

Have you checked to see if the dashboard pod is actually running? If there are no endpoints, then the pod probably hasn't started yet, or is crashing.

`kubectl get pod,endpoints -A -o wide`

**Reply by ivdok** [reply]
```E0407 14:01:45.128187  839617 memcache.go:287] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0407 14:01:45.535865  839617 memcache.go:121] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0407 14:01:45.673650  839617 memcache.go:121] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0407 14:01:45.812121  839617 memcache.go:121] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0407 14:01:45.952304  839617 memcache.go:121] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0407 14:01:46.091514  839617 memcache.go:121] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
NAMESPACE              NAME                                            READY   STATUS             RESTARTS        AGE     IP          NODE   NOMINATED NODE   READINESS GATES
kube-system            pod/coredns-7c444649cb-czpcf                    0/1     Running            0               8m44s   10.42.0.3   k8s    <none>           <none>
kubernetes-dashboard   pod/dashboard-metrics-scraper-7bc864c59-qfzs8   1/1     Running            0               7m41s   10.42.0.8   k8s    <none>           <none>
kube-system            pod/metrics-server-7b67f64457-r6x95             0/1     CrashLoopBackOff   5 (2m41s ago)   8m44s   10.42.0.5   k8s    <none>           <none>
kube-system            pod/local-path-provisioner-5d56847996-2sh7r     0/1     CrashLoopBackOff   5 (2m42s ago)   8m44s   10.42.0.4   k8s    <none>           <none>
kube-system            pod/helm-install-traefik-crd-kcfpl              1/1     Running            2 (2m17s ago)   8m44s   10.42.0.2   k8s    <none>           <none>
kube-system            pod/helm-install-traefik-84hz8                  1/1     Running            2 (2m17s ago)   8m44s   10.42.0.6   k8s    <none>           <none>
kubernetes-dashboard   pod/kubernetes-dashboard-6c7ccbcf87-7fppz       0/1     CrashLoopBackOff   5 (85s ago)     7m41s   10.42.0.7   k8s    <none>           <none>

NAMESPACE              NAME                                  ENDPOINTS            AGE
default                endpoints/kubernetes                  10.99.0.254:6443   8m59s
kube-system            endpoints/kube-dns                                         8m44s
kube-system            endpoints/metrics-server                                   8m44s
kubernetes-dashboard   endpoints/dashboard-metrics-scraper   10.42.0.8:8000       7m41s
kubernetes-dashboard   endpoints/kubernetes-dashboard                             7m41s
```

By all means, this IS a nondescript error. I followed the guide, and k3s craps itself for no reason.

**Reply by brandond** [reply]
OK, so your next step is to look at logs for the pods that are in state CrashLoopBackoff and see why they're not working properly. In addition to the two dashboard pods, the local-path-provisioner pod is also crashing for some reason. 

That is a quite good message as far as Kubernetes is concerned. It doesn't know **why** the service doesn't have any endpoints, it just knows that it doesn't, and therefore there is nothing to connect to. It's up to you as the administrator to figure out why, and the most likely reason is that the pods that back it are not ready.
